python/ray/tune/tests/execution/test_controller_resources_integration.py

```diff
@@ -163,18 +164,15 @@ def test_resources_changing(ray_start_4_cpus_2_gpus_extra, resource_manager_cls)
     """
 
     class ChangingScheduler(FIFOScheduler):
-        def __init__(self):
-            self._has_received_one_trial_result = False
-
-        # For figuring out how many runner.step there are.
-        def has_received_one_trial_result(self):
-            return self._has_received_one_trial_result
-
         def on_trial_result(self, tune_controller, trial, result):
             if result["training_iteration"] == 1:
-                self._has_received_one_trial_result = True
-                tune_controller.pause_trial(trial)
+                # NOTE: This is a hack to get around the new pausing logic,
+                # which doesn't set the trial status to PAUSED immediately.
+                orig_status = trial.status
+                trial.set_status(Trial.PAUSED)
                 trial.update_resources(dict(cpu=4, gpu=0))
+                trial.set_status(orig_status)
+                return TrialScheduler.PAUSE
```

Comment:
I've updated the test to pass more minimally, but calling `pause_trial` within `on_trial_result`  is still an issue.